sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -140,33 +140,29 @@ def start(self, create_sql_context=True):
             html = get_sessions_info_html([self], self.id)
             self.ipython_display.html(html)
 
-            if create_sql_context:
-                self.create_sql_context()
+            command = Command("spark")
+            (success, out) = command.execute(self)
+
+            if success:
+                self.logger.debug(u"SparkSession exists for session as variable spark.".format(self.kind))
+                self.ipython_display.writeln(u"SparkSession exists for session as variable spark...")
+                self.context_variable_name = "spark"
+            else:
+                command = Command("sqlContext")
+                (success, out) = command.execute(self)
+                if success:
+                    self.logger.debug(u"HiveContext exists for session as variable sqlContext.".format(self.kind))
+                    self.ipython_display.writeln(u"HiveContext exists for session as variable sqlContext...")
+                    self.context_variable_name = "sqlContext"
+                else:
+                    raise SqlContextNotFoundException(u"Neither SparkSession nor HiveContext is found")
```

Comment:
just say neither sqlContext nor spark variable is found. sqlContext can really be SQLContext or HiveContext.